File: qianbot-frontend/src/api/userController.ts
// @ts-ignore
/* eslint-disable */
import request from '@/utils/request';

/** getUserById GET /api/user/get */
export async function getUserByIdUsingGet(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.getUserByIdUsingGETParams,
  options?: { [key: string]: any },
) {
  return request<API.BaseResponseUser_>('/api/user/get', {
    method: 'GET',
    params: {
      ...params,
    },
    ...(options || {}),
  });
}

/** getLoginUser GET /api/user/get/login */
export async function getLoginUserUsingGet(options?: { [key: string]: any }) {
  return request<API.BaseResponseLoginUserVO_>('/api/user/get/login', {
    method: 'GET',
    ...(options || {}),
  });
}

/** getUserVOById GET /api/user/get/vo */
export async function getUserVoByIdUsingGet(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.getUserVOByIdUsingGETParams,
  options?: { [key: string]: any },
) {
  return request<API.BaseResponseUserVO_>('/api/user/get/vo', {
    method: 'GET',
    params: {
      ...params,
    },
    ...(options || {}),
  });
}

/** listUserByPage POST /api/user/list/page */
export async function listUserByPageUsingPost(
  body: API.UserQueryRequest,
  options?: { [key: string]: any },
) {
  return request<API.BaseResponsePageUser_>('/api/user/list/page', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    data: body,
    ...(options || {}),
  });
}

/** listUserVOByPage POST /api/user/list/page/vo */
export async function listUserVoByPageUsingPost(
  body: API.UserQueryRequest,
  options?: { [key: string]: any },
) {
  return request<API.BaseResponsePageUserVO_>('/api/user/list/page/vo', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    data: body,
    ...(options || {}),
  });
}

/** userLogin POST /api/user/login */
export async function userLoginUsingPost(
  body: API.UserLoginRequest,
  options?: { [key: string]: any },
) {
  return request<API.BaseResponseLoginUserVO_>('/api/user/login', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    data: body,
    ...(options || {}),
  });
}

/** userLogout GET /api/user/logout */
export async function userLogoutUsingGet(options?: { [key: string]: any }) {
  return request<API.BaseResponseBoolean_>('/api/user/logout', {
    method: 'GET',
    ...(options || {}),
  });
}

/** userRegister POST /api/user/register */
export async function userRegisterUsingPost(
  body: API.UserRegisterRequest,
  options?: { [key: string]: any },
) {
  return request<API.BaseResponseLong_>('/api/user/register', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    data: body,
    ...(options || {}),
  });
}

/** userRetryPassword POST /api/user/retry/password */
export async function userRetryPasswordUsingPost(
  body: API.UserUpdateRequest,
  options?: { [key: string]: any },
) {
  return request<API.BaseResponseBoolean_>('/api/user/retry/password', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    data: body,
    ...(options || {}),
  });
}

/** getPasswordCode GET /api/user/send/code/password */
export async function getPasswordCodeUsingGet(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.getPasswordCodeUsingGETParams,
  options?: { [key: string]: any },
) {
  return request<API.BaseResponseBoolean_>('/api/user/send/code/password', {
    method: 'GET',
    params: {
      ...params,
    },
    ...(options || {}),
  });
}

/** getRegisterCode GET /api/user/send/code/register */
export async function getRegisterCodeUsingGet(
  // 叠加生成的Param类型 (非body参数swagger默认没有生成对象)
  params: API.getRegisterCodeUsingGETParams,
  options?: { [key: string]: any },
) {
  return request<API.BaseResponseBoolean_>('/api/user/send/code/register', {
    method: 'GET',
    params: {
      ...params,
    },
    ...(options || {}),
  });
}

/** userUpdate POST /api/user/update */
export async function userUpdateUsingPost(
  body: API.UserUpdateRequest,
  options?: { [key: string]: any },
) {
  return request<API.BaseResponseBoolean_>('/api/user/update', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    data: body,
    ...(options || {}),
  });
}

File: qianbot-backend/src/main/java/com/qian/qianbotbackend/controller/user/UserController.java
package com.qian.qianbotbackend.controller.user;

import com.baomidou.mybatisplus.extension.plugins.pagination.Page;
import com.qian.qianbotbackend.annotation.AuthCheck;
import com.qian.qianbotbackend.common.BaseResponse;
import com.qian.qianbotbackend.common.ErrorCode;
import com.qian.qianbotbackend.common.ResultUtils;
import com.qian.qianbotbackend.constant.UserConstant;
import com.qian.qianbotbackend.exception.BusinessException;
import com.qian.qianbotbackend.manager.EmailManager;
import com.qian.qianbotbackend.model.user.domain.User;
import com.qian.qianbotbackend.model.user.dto.user.UserLoginRequest;
import com.qian.qianbotbackend.model.user.dto.user.UserQueryRequest;
import com.qian.qianbotbackend.model.user.dto.user.UserRegisterRequest;
import com.qian.qianbotbackend.model.user.dto.user.UserUpdateRequest;
import com.qian.qianbotbackend.model.user.vo.LoginUserVO;
import com.qian.qianbotbackend.model.user.vo.UserVO;
import com.qian.qianbotbackend.service.UserService;
import lombok.extern.slf4j.Slf4j;
import org.springframework.web.bind.annotation.*;

import javax.annotation.Resource;
import javax.servlet.http.HttpServletRequest;

@RestController
@RequestMapping("/user")
@Slf4j
public class UserController {
    @Resource
    private UserService userService;

    @Resource
    private EmailManager emailManager;

    // region 用户

    /**
     * 用户注册
     *
     * @param userRegisterRequest 用户注册请求
     * @return id
     */
    @PostMapping("/register")
    public BaseResponse<Long> userRegister(@RequestBody UserRegisterRequest userRegisterRequest) {
        return ResultUtils.success(userService.userRegister(userRegisterRequest));
    }

    /**
     * 用户登录
     *
     * @param userLoginRequest 用户登录请求
     * @param request          请求
     * @return 登录用户信息
     */
    @PostMapping("/login")
    public BaseResponse<LoginUserVO> userLogin(@RequestBody UserLoginRequest userLoginRequest, HttpServletRequest request) {
        return ResultUtils.success(userService.userLogin(userLoginRequest, request));
    }

    /**
     * 用户重置密码
     *
     * @param userUpdateRequest 请求
     * @return 是否成功
     */
    @PostMapping("/retry/password")
    public BaseResponse<Boolean> userRetryPassword(@RequestBody UserUpdateRequest userUpdateRequest, HttpServletRequest request) {
        userService.retryPassword(userUpdateRequest, request);
        return ResultUtils.success(true);
    }

    /**
     * 用户更新
     *
     * @param userUpdateRequest 请求
     * @return 是否成功
     */
    @PostMapping("/update")
    @AuthCheck(mustRole = UserConstant.ROLE_USER)
    public BaseResponse<Boolean> userUpdate(@RequestBody UserUpdateRequest userUpdateRequest) {
        userService.userUpdate(userUpdateRequest);
        return ResultUtils.success(true);
    }

    /**
     * 用户注销
     *
     * @param request 请求
     * @return 是否成功
     */
    @GetMapping("/logout")
    public BaseResponse<Boolean> userLogout(HttpServletRequest request) {
        return ResultUtils.success(userService.userLogout(request));
    }

    /**
     * 获取当前登录用户
     *
     * @param request 请求
     * @return 登录用户信息
     */
    @GetMapping("/get/login")
    public BaseResponse<LoginUserVO> getLoginUser(HttpServletRequest request) {
        return ResultUtils.success(userService.getLoginUserVO(userService.getLoginUser(request)));
    }
    // endregion

    // region 增删改查

    /**
     * 根据 id 获取用户（仅管理员）
     *
     * @param id 用户id
     * @return 用户信息
     */
    @GetMapping("/get")
    @AuthCheck(mustRole = UserConstant.ROLE_ADMIN)
    public BaseResponse<User> getUserById(long id) {
        return ResultUtils.success(userService.getUserById(id));
    }

    /**
     * 根据 id 获取包装类
     *
     * @param id 用户id
     * @return userVO
     */
    @GetMapping("/get/vo")
    public BaseResponse<UserVO> getUserVOById(long id) {
        return ResultUtils.success(userService.getUserVO(userService.getUserById(id)));
    }

    /**
     * 分页获取用户列表（仅管理员）
     *
     * @param userQueryRequest 分页请求
     * @return 用户列表
     */
    @PostMapping("/list/page")
    @AuthCheck(mustRole = UserConstant.ROLE_ADMIN)
    public BaseResponse<Page<User>> listUserByPage(@RequestBody UserQueryRequest userQueryRequest) {
        return ResultUtils.success(userService.listUserByPage(userQueryRequest));
    }

    /**
     * 分页获取用户封装列表
     *
     * @param userQueryRequest 分页请求
     * @return 用户封装列表
     */
    @PostMapping("/list/page/vo")
    public BaseResponse<Page<UserVO>> listUserVOByPage(@RequestBody UserQueryRequest userQueryRequest) {
        return ResultUtils.success(userService.listUserVOByPage(userQueryRequest));
    }
    // endregion

    // region 验证码

    /**
     * 获取注册验证码
     *
     * @param email 邮箱
     * @return 是否成功
     */
    @GetMapping("/send/code/register")
    public BaseResponse<Boolean> getRegisterCode(String email) {
        userService.verifyEmail(email, true);
        String code = emailManager.genAndSaveCode(email);
        emailManager.sendCodeForRegister(email, code);
        return ResultUtils.success(true);
    }

    /**
     * 获取密码重置验证码
     *
     * @param email 邮箱
     * @return 是否成功
     */
    @GetMapping("/send/code/password")
    public BaseResponse<Boolean> getPasswordCode(String email) {
        userService.verifyEmail(email, false);
        String code = emailManager.genAndSaveCode(email);
        emailManager.sendCodeForPassword(email, code);
        return ResultUtils.success(true);
    }
    // endregion
}
